src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py

```diff
@@ -118,7 +118,7 @@ def __init__(self, attn_res=16):
         self.step_store = self.get_empty_store()
         self.attention_store = {}
         self.curr_step_index = 0
-        self.attn_res = attn_res
+        self.attn_res = int(np.ceil(width / 32)), int(np.ceil(height / 32))
```

Comment:
Can we make this a variable with a default value `32` instead of hardcode it?  It won't be a common use case, but someone might want to use attention maps from a different layer when using this pipeline with a different model 

also I think this might break the current tests, the expected results in the tests are based on attention resolution as `16x16`, and sample size `32 x 32`, with this code change the attention resolution become `8x8` and I think it won't work - once we make `32` a variable we can adjust the test accordingly 

https://github.com/huggingface/diffusers/blob/main/tests/pipelines/stable_diffusion_2/test_stable_diffusion_attend_and_excite.py#L43

Reply:
> with this code change the attention resolution become 8x8 and I think it won't work 
Note, ``512 / 32 = 16`` not ``8``.  You would only get 8x8 with an 128x128 output size.

Rather than add a new variable with default value 32 it might be better to restore the ``attn_res`` option but as a 2D variable with ``None`` as the default.  If it's ``None`` it will be automatically replaced with ``ceil(width / 32), ceil(height / 32)``.  Otherwise, explaining what the 32 value means is a bit complicated.

Reply:
@jcoffland sounds good!

Reply:
Updates made.